server/src/main/java/com/genymobile/scrcpy/Device.java

```diff
@@ -22,11 +23,13 @@
 
     private final ServiceManager serviceManager = new ServiceManager();
 
+    private final DisplayInfo displayInfo;
```

Comment:
The `DisplayInfo` data is not valid after initialization (for example, the `rotation` field is obsolete after a rotation). Therefore, I think it should not be kept for the lifetime of the `Device` instance.

Maybe only its relevant fields (used in this class) could be computed initially and kept (for example `supportsInputEvents`, `layerStack`…)?